app/forms/VATNumberFormProvider.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import forms.mappings.Mappings
+import play.api.data.Form
+import play.api.data.Forms.single
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class VATNumberFormProvider  @Inject() extends FormErrorHelper with Mappings {
+
+  val vatNumberKey = "vatNumber"
+  val errorKeyRoot = s"vatNumber.error"
+  val vatNumberRequired = s"$errorKeyRoot.togc.required"
+  val tooManyChars = s"$errorKeyRoot.tooMany"
+  val nonNumeric = s"$errorKeyRoot.nonNumeric"
+  val invalid = s"$errorKeyRoot.invalid"
+  val regex = """^[0-9]{9,9}$"""
```

Comment:
I don't think you need {9,9} in the regex, just {9} should suffice. Odd that the doc says {1,9} as VRNs are all 9 digit long